lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -751,15 +769,36 @@ migrateByronWallet _rndCtx _seqCtx _sourceWid _targetWid _migrateData =
     throwError err501
 
 listByronWallets
-    :: ctx
+    :: forall s t k. (DefineTx t, s ~ RndState t)
+    => ApiLayer s t k
     -> Handler [ApiByronWallet]
-listByronWallets _ = throwError err501
+listByronWallets ctx = do
+    wids <- liftIO $ Registry.keys re
+    fmap fst . sortOn snd <$>
+        mapM (getWalletWithCreationTime mkApiByronWallet ctx) (ApiT <$> wids)
+  where
+    re = ctx ^. workerRegistry @s @t @k
 
 postByronWallet
-    :: ctx
+    :: forall t. (DefineTx t, KeyToAddress t RndKey)
+    => ApiLayer (RndState t) t RndKey
     -> ByronWalletPostData
     -> Handler ApiByronWallet
-postByronWallet _ _ = throwError err501
+postByronWallet ctx body = do
+    void
+        . liftHandler
+        . createWallet ctx wid name
+        . mkRndState rootXPrv
+        =<< liftIO (getStdRandom random)
```

Comment:
just nitpicking -  this supposed to increase readability? :nerd_face:

Reply:
yeah, I know how it flows ;-) And as I read from left to right, this structure forced me to do the opposite :-)

Reply:
> yeah, I know how it flows ;-) And as I read from left to right, this structure forced me to do the opposite :-)

Fair enough! But then surely the same applies to any composition of functions:

```hs
foo :: SomeComplicatedType
foo = f . g . h
```

In the above example, data flows (conceptually) from right to left.

Reply:
I tend to agree with @paweljakubas on that one :grimacing: ... I don't mind much the function composition, but the bind `>>=` makes the whole not very readable IMO.